### PR TITLE
feature/login panel

### PR DESCRIPTION
**Pull request by veroniqaa**
FEAT(#9): updating login panel and adding tests

### CODE REVIEW

File: front/src/components/LoginPanel/LoginPanel.test.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        ).container;
+    });
+
+    it("should render login panel by default", () => {
+        let loginForm = screen.getByTestId("loginForm");
+        expect(loginForm).toBeInTheDocument();
+    });
+
+    it("should have two buttons to change subpanels", () => {
+        let buttonsDiv = container.querySelector(".buttons");
```
Comment (frogi16) [inline front/src/components/LoginPanel/LoginPanel.test.js:23]: Querying by div or other properties invisible for user is the least preferable option, but in this case is necessary to distinguish from buttons rendered by nested components.
Consider changing class name to something specific to this and only this div or even adding some other easily-distinguishable property (role, id, etd).

File: front/src/components/LoginPanel/LoginPanel.test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        expect(buttonsDiv).not.toBeNull();
+        expect(within(buttonsDiv).getAllByRole("button")).toHaveLength(2);
+    });
+
+    it("should render register panel after clicking appropriate button", () => {
+        let registerButton = screen.getByRole("button", {
+            name: /zarejestruj się/i,
```
Comment (frogi16) [inline front/src/components/LoginPanel/LoginPanel.test.js:30]: If to search specifically for button, add this info to query and let react testing library do its job :)

File: front/src/components/LoginPanel/comp/LoginForm.test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    it("should have one text input", () => {
+        let textInputs = screen.getAllByRole("textbox");
+        expect(textInputs).toHaveLength(1);
+    });
+
+    it("should have one button", () => {
+        let buttons = screen.getAllByRole("button");
+        expect(buttons).toHaveLength(1);
+    });
+});
```
Comment (frogi16) [inline front/src/components/LoginPanel/comp/LoginForm.test.js:29]: Rewrite this file in a way I did with LoginPanel.test.js in commit fb558db

**Review by frogi16 — CHANGES_REQUESTED** [review]

File: front/src/components/LoginPanel/comp/LoginForm.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
-import React, { useState, useEffect } from "react";
+import React, { useState } from "react";
+import "./LoginForm.css";
 import Button from "../../UI/Button/Button";
 import { gql, useMutation } from "@apollo/client";
```
Comment (frogi16) [inline front/src/components/LoginPanel/comp/LoginForm.js:5]: Lines 18-22:
```
    const [loginUser, { data }] = useMutation(LOGIN_USER, {
        onError(err) {
            console.log(err);
        },
    });
```
Change this to something visible to user, even alert would be better.

File: front/src/components/LoginPanel/comp/LoginForm.test.js [hunk 5]
```diff
@@ -1,29 +1,26 @@
...
     beforeEach(() => {
-        render(
-            <ApolloProvider client={client}>
+        container = render(
+            <MockedProvider mocks={[]} addTypename={false}>
                 <LoginForm />
-            </ApolloProvider>
-        );
+            </MockedProvider>
+        ).container;
```
Comment (frogi16) [inline front/src/components/LoginPanel/comp/LoginForm.test.js]: You don't use container anywhere.

File: front/src/components/LoginPanel/comp/LoginForm.test.js [hunk 6]
```diff
@@ -1,29 +1,26 @@
...
     });
 
     it("should have one text input", () => {
-        let textInputs = screen.getAllByRole("textbox");
-        expect(textInputs).toHaveLength(1);
+        let textInput = screen.getByRole("textbox");
+        expect(textInput).toBeInTheDocument();
     });
```
Comment (frogi16) [inline front/src/components/LoginPanel/comp/LoginForm.test.js]: Search also by name

File: front/src/components/LoginPanel/comp/LoginForm.test.js [hunk 7]
```diff
@@ -1,29 +1,26 @@
...
 
     it("should have one button", () => {
-        let buttons = screen.getAllByRole("button");
-        expect(buttons).toHaveLength(1);
+        let loginButton = screen.getByRole("button", {
+            name: /zaloguj się/i,
+        });
+        expect(loginButton).toBeInTheDocument();
```
Comment (frogi16) [inline front/src/components/LoginPanel/comp/LoginForm.test.js:24]: Remove "się" in case text will ever change

**Review by frogi16 — CHANGES_REQUESTED** [review]

File: front/src/components/LoginPanel/comp/LoginForm.js [hunk 8]
```diff
@@ -40,29 +41,33 @@ const LoginForm = (props) => {
...
+                    onChange={handleLoginChange}
+                />{" "}
+                <br />
+                <input
+                    type="password"
+                    placeholder="Hasło"
+                    className="form-control login-input"
+                    value={password}
+                    onChange={handlePasswordChange}
+                />{" "}
```
Comment (frogi16) [inline front/src/components/LoginPanel/comp/LoginForm.js:61]: Use autoComplete like in 50b47ce.

File: front/src/components/LoginPanel/comp/LoginForm.js [hunk 9]
```diff
@@ -58,6 +60,7 @@ const LoginForm = (props) => {
                     className="form-control login-input"
                     value={password}
                     onChange={handlePasswordChange}
+                    autoComplete="new-password"
```
Comment (frogi16) [inline front/src/components/LoginPanel/comp/LoginForm.js:63]: Shouldn't it be?
```suggestion
                    autoComplete="current-password"
```

File: front/src/components/LoginPanel/comp/LoginForm.test.js [hunk 10]
```diff
@@ -3,23 +3,29 @@ import { MockedProvider } from "@apollo/client/testing";
...
             <MockedProvider mocks={[]} addTypename={false}>
                 <LoginForm />
             </MockedProvider>
-        ).container;
+        );
     });
 
-    it("should have one text input", () => {
-        let textInput = screen.getByRole("textbox");
+    it("should have a textbox input for login or email", () => {
```
Comment (frogi16) [inline front/src/components/LoginPanel/comp/LoginForm.test.js:14]: I don't think "or email" is tested here. Update test name

**Review by frogi16 — CHANGES_REQUESTED** [review]

**Review by frogi16 — APPROVED** [review]

**Review by pieczonkam — APPROVED** [review]

**Pull request merged**